### PR TITLE
[Docs] Better explain the modular approach.

### PR DESCRIPTION
**Pull request by chriseth**
Refs https://github.com/ethereum/solidity/issues/6784

### CODE REVIEW

**Review by guylando — CHANGES_REQUESTED** [review]
"A modular approach to building your contracts helps you reduce the complexity. If you specify..." -> "A modular approach to building your contracts helps you reduce the complexity and improve the readability which will help to identify bugs and vulnerabilities during development and code review. Furthermore, when you specify..."

File: docs/examples/modular.rst [hunk 1]
```diff
@@ -4,10 +4,16 @@
 Modular Contracts
 *****************
 
-A modular approach to building your contracts helps you prevent overflow risks
-and unexpected tokens. In the example below, the contract uses the ``move`` method
+A modular approach to building your contracts helps you reduce the complexity.
+If you specify and control the behaviour or each module in isolation, the
+interactions you have to consider are only those between the specifications
```
Comment (bshastry) [inline docs/examples/modular.rst]: `between the module specifications` instead of `between the specifications`?

File: docs/examples/modular.rst [hunk 2]
```diff
@@ -4,10 +4,16 @@
...
+and not every other moving part of the contract.
+In the example below, the contract uses the ``move`` method
 of the ``Balances`` :ref:`library <libraries>` to check that balances sent between
-addresses match what you expect.
+addresses match what you expect. In this way, the ``Balances`` library
+provides an isolated component that properly tracks balances of accounts.
+It is easy to verify that it never produces negative balances or overflows
```
Comment (bshastry) [inline docs/examples/modular.rst]: `It is easy to verify that the ``Balances`` library never produces negative balances...`? (replaced it by the Balances library)

File: docs/examples/modular.rst [hunk 3]
```diff
@@ -4,10 +4,16 @@
...
 *****************
 
-A modular approach to building your contracts helps you prevent overflow risks
-and unexpected tokens. In the example below, the contract uses the ``move`` method
+A modular approach to building your contracts helps you reduce the complexity.
+If you specify and control the behaviour or each module in isolation, the
+interactions you have to consider are only those between the specifications
+and not every other moving part of the contract.
+In the example below, the contract uses the ``move`` method
 of the ``Balances`` :ref:`library <libraries>` to check that balances sent between
```
Comment (bshastry) [inline docs/examples/modular.rst:14]: Not related to this PR: Could this and other references be reformaated as `hyperlink reference <URI>`_ so as to not interrupt the flow of text. Note that the `:ref: `library <libraries>` is rendered as-is.

**Comment by chriseth** [comment]
Updated.

**Review by bshastry — APPROVED** [review]

**Pull request merged**